spec/migration/parity_check/dynamic_request_content_spec.rb

```diff
@@ -159,6 +197,68 @@
       end
     end
 
+    context "when fetching `active_participant_defer_body`" do
+      let(:identifier) { :active_participant_defer_body }
+      let(:lead_provider_delivery_partnership) { FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider:) }
+      let!(:school_partnership) { FactoryBot.create(:school_partnership, lead_provider_delivery_partnership:) }
+      let!(:training_period) { FactoryBot.create(:training_period, :for_ect, :ongoing, :active, school_partnership:) }
+      let(:teacher) { training_period.trainee.teacher }
+
+      before do
+        # Deferred participant for current lead provider
+        FactoryBot.create(:training_period, :for_ect, :ongoing, :deferred, school_partnership:)
+        # Withdrawn participant for current lead provider
+        FactoryBot.create(:training_period, :for_ect, :ongoing, :withdrawn, school_partnership:)
+        # Active participant for different lead provider
+        FactoryBot.create(:training_period, :for_ect, :ongoing, :active)
+        # Stub deferral reasons so we have a predictable value
+        allow(TrainingPeriod).to receive(:deferral_reasons).and_return({ parental_leave: "moved_school" })
+      end
+
+      it "returns a participant defer body" do
+        expect(fetch).to eq({
+          data: {
+            type: "participant-defer",
+            attributes: {
+              reason: "moved-school",
+              course_identifier: "ecf-induction"
+            },
+          },
+        })
+      end
+    end
+
+    context "when fetching `deferred_participant_defer_body`" do
+      let(:identifier) { :deferred_participant_defer_body }
+      let(:lead_provider_delivery_partnership) { FactoryBot.create(:lead_provider_delivery_partnership, active_lead_provider:) }
+      let!(:school_partnership) { FactoryBot.create(:school_partnership, lead_provider_delivery_partnership:) }
+      let!(:training_period) { FactoryBot.create(:training_period, :for_ect, :ongoing, :deferred, school_partnership:) }
+      let(:teacher) { training_period.trainee.teacher }
+
+      before do
+        # Active participant for current lead provider
+        FactoryBot.create(:training_period, :for_ect, :ongoing, :active, school_partnership:)
```

Comment:
I think some of these should be mentors as what we're really testing is it gives the correct `course_identifier`?